code-challenges.test.js

```diff
@@ -10,43 +10,169 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
+/*
+Boiler Plate
+describe("areYouHungry", () => {
+it("returns eat food or keep coding based on input", () => {
+    expect(areYouHungry("yes")).toEqual("eat food")
+    expect(areYouHungry("no")).toEqual("keep coding")
+ })
+})
+*/
+
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("secretCode", () => {
+    it("returns a string with a coded message", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(secretCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(secretCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(secretCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+     })
+    })
+
+    // ReferenceError: secretCode is not defined
 
 // b) Create the function that makes the test pass.
 
+/*
+Create a function that takes one parameter named string.
+The single parameter will take in a string as an argument.
+Find a way to iterate through the string.
+Select the character inside of the string.
+Replace the character with an assigned number.
+Return the encrypted string that has the same length as the original string.
+*/
+
+// const secretCode = (string) => {
+//     let arr = [...string]
+//     return arr
+// }
+//The initial troubleshooting step was to confirm that I converted the string into an array.
+// Expected: "L4ck4d41s1c4l"
+//     Received: ["L", "a", "c", "k", "a", "d", "a", "i", "s", "i", …]
+//     Test Suites: 1 failed, 1 total
+//     Tests:       1 failed, 1 total
+
+const secretCode = (string) => {
+    let four = /a/ig
+    let three = /e/ig
+    let one = /i/ig
+    let zero = /o/ig
+    return string.replaceAll(four, 4).replaceAll(three, 3).replaceAll(one, 1).replaceAll(zero, 0)
+}
+/*
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+I looked up if there was a way to replace a character with a number. 
+My search led me to the built-in method .replaceAll()
+.replaceAll() returns a string with all matches of a pattern that can be replaced by a replacement. 
+The pattern being used is regex, or regular expressions, in order to find each instance of the argument and replaced with a number.
+I am sure there is a way to make this code efficient, but I am unable to figure out a way to solve that issue.
```

Comment:
This looks really efficient already and is one of my favorite approaches for this problem.  One way to shorten this up just a little bit would be to place the regex inside replaceAll instead of assigning it to a variable.  Ex: `string.replaceAll(/a/ig, 4)`